static/js/stream_edit.js

```diff
@@ -489,6 +490,13 @@ export function initialize() {
         }
 
         stream_settings_ui.sub_or_unsub(sub);
+        // Refresh the view only if the stream is unsubscribed before
+        // and subscribed now
+        if (!sub.subscribed) {
+            narrow.activate([{operator: "stream", operand: sub.name}], {
+                trigger: "subscribe button",
```

Comment:
User can subscribe to the stream when narrowed to a topic too. We should handle that too, which means the narrowed view should be same as before with only change that latest messages are shown.